### PR TITLE
[xbmc] refactor instantiation of storage provider and power manager syscalls

### PR DESCRIPTION
**Pull request by afedchin**
this follows tendantion for removing ifdefs and implementing design patterns

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/powermanagement/IPowerSyscall.h [hunk 1]
```diff
@@ -64,6 +64,12 @@ class IPowerSyscall
   virtual bool PumpPowerEvents(IPowerEventsCallback *callback) = 0;
 
   static const int MAX_COUNT_POWER_FEATURES = 4;
+
+  /**\brief Called by power manager to create platform power system adapter
+  *
+  * This method used to create platfrom specified power system adapter
+  */
+  static IPowerSyscall* CreateInstance();
```
Comment (FernetMenta) [inline xbmc/powermanagement/IPowerSyscall.h]: why not having platforms register a create function? as far as I can see this is an optional component.

Comment (afedchin) [reply xbmc/powermanagement/IPowerSyscall.h]: @FernetMenta I've updated this, but I'm not sure that I did this right.

Comment (FernetMenta) [reply xbmc/powermanagement/IPowerSyscall.h]: @afedchin I would drop CNullPowerSyscall and check m_instance instead

**Comment by afedchin** [comment]
jenkins build this please

File: xbmc/powermanagement/IPowerSyscall.h (hunk 1 above)
Comment (afedchin) [reply xbmc/powermanagement/IPowerSyscall.h]: @FernetMenta done. please check. is windowing right place to register those things?

Comment (FernetMenta) [reply xbmc/powermanagement/IPowerSyscall.h]: @afedchin should do for now. Maybe we find a better place later.

**Comment by afedchin** [comment]
manual build [169](http://jenkins.kodi.tv/view/All/job/BuildMulti-PR-Manually/169/) successes
if no objections I'll merge this tonight

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by popcornmix** [comment]
It builds and runs on Pi.
My environment only includes "Exit" in the shutdown menu and that is unchanged.
I know LE additionally includes shutdown/reboot, but I've not tested that yet.
(although I've included this in my tree for tonight's milhouse build).

**Comment by popcornmix** [comment]
No obvious issue in milhouse build - shutdown and reboot still worked, so no objections here.

**Comment by afedchin** [comment]
thanks for testing.

**Pull request merged**